src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs

```diff
@@ -83,6 +83,7 @@ public void TrimmingOptions_Are_Defaulted_Correctly_On_Aot_Apps(string targetFra
 
             responseFileContents.Should().Contain("--feature:System.Text.Json.JsonSerializer.IsReflectionEnabledByDefault=false");
             responseFileContents.Should().Contain("--feature:System.Diagnostics.Tracing.EventSource.IsSupported=true");
+            responseFileContents.Should().Contain("--runtimeknob:System.GC.DynamicAdaptationMode=1");
```

Comment:
it's literally called `--runtimeknob`? 😆

Reply:
https://github.com/dotnet/runtime/blob/3afe6bd73788bd941dc06e4160616dc100bbcc43/src/coreclr/tools/aot/ILCompiler/ILCompilerRootCommand.cs#L95-L96

```C#
        public Option<string[]> RuntimeKnobs { get; } =
            new(new[] { "--runtimeknob" }, Array.Empty<string>, "Runtime knobs to set");
```

😆

Reply:
In CoreCLR lingo, the settings one can set through environment variables are "config values" (e.g. `DOTNET_gcServer=1`). The values in runtimeconfig.json are "knob values" (e.g. `System.GC.Server=true`).

I can't say I'm thrilled to see the existence/name/contents of the RSP file interpreted outside of the dotnet/runtime repo though.

Reply:
> I can't say I'm thrilled to see the existence/name/contents of the RSP file interpreted outside of the dotnet/runtime repo though.

Agreed this isn't ideal. But I can't really think of a better way to ensure the properties are flown correctly. Running the app, giving it load, and inspecting its memory usage would be the "real" way to test it. But that's too heavy handed.

The good news is that this is limited to a very small subset of tests (namely 1) to ensure we have the right defaulting logic in place. If we ever change these ilc command line argument names, it is simple to fix it here.

Reply:
> Agreed this isn't ideal. But I can't really think of a better way to ensure the properties are flown correctly. Running the app, giving it load, and inspecting its memory usage would be the "real" way to test it. But that's too heavy handed.

If we're allowed to run the app, we also inspect AppContext (these also flow to AppContext), or there might be a GC API we can query directly. Similarly for the feature switches.

I'm not saying we need to do this because digging into RSP also works and this code is already written, but maybe for next time.